data-protocols/ids/ids-api-multipart-dispatcher-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/sender/IdsMultipartSender.java

```diff
@@ -312,4 +323,20 @@ private IdsMultipartParts extractResponseParts(ResponseBody body) throws Excepti
                 .build();
     }
 
+    private R checkResponseType(R response) {
+        var type = getAllowedResponseTypes();
+
+        Objects.requireNonNull(type);
```

Comment:
as `getAllowedResponseTypes` result is always hardcoded, I would assume that's always non-null, maybe the `@NotNull` annotation could be used on the method and this guard avoided.